### ISSUE TITLE
Sanitize objects before DeserliazeObjects

### ISSUE THREAD

**Issue by RaduCStefanescu** [code4romania/stam-acasa#421]
Potential Remote Code Execution (RCE) using DeserializeObject function on arbitrary user provided data in `backend/src/StamAcasa.Api/Controllers/FormController.cs`

**Comment by andreea-anastasescu** [comment]
il pot lua eu

**Comment by andreea-anastasescu** [comment]
I have looked into this reported issue.
According to:
 * https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2330?view=vs-2019
 * https://www.alphabot.com/security/blog/2017/net/How-to-configure-Json.NET-to-create-a-vulnerable-web-API.html
 * https://stackoverflow.com/questions/49038055/external-json-vulnerable-because-of-json-net-typenamehandling-auto/49040862
 
 What I understood is that the main threat is deserializing with newtonsoft.Json.JsonConvert with TypeNameHandling different then None.
 Currently no settings is done on deserialization, and the default for TypeNameHandling is 'None' - so this should be safe.

I see nothing here to be changed for security purposes. 
Maybe i missed something - in this case please advise

**Comment by idormenco** [comment]
@andreea-anastasescu ,you are right. Most probably they found this vulnerability using some static code analysis tool and most probably this vulnerability may not affect us (don't quote me on  this please). 
What we can do is to improve our API design and not rely on dynamic stuff. We could come up with a model which will be submitted by frontend .

Something like this :
```
public class UserForm
    {
        [JsonProperty("formId")]
        public long FormId { get; set; }

        [JsonProperty("timestamp")]
        public long Timestamp { get; set; }

        [JsonProperty("answers")]
        public Answer[] Answers { get; set; }
    }

    public class Answer
    {
        [JsonProperty("id")]
        public long Id { get; set; }

        [JsonProperty("questionText")]
        public string QuestionText { get; set; }

        [JsonProperty("answer", NullValueHandling = NullValueHandling.Ignore)]
        public string AnswerAnswer { get; set; }
    }
```

This will help us also when we will implement validation of user answers.

If you have any questions do not hesitate to ask here or on slack.

**Comment by andreea-anastasescu** [comment]
I will use then the above mentioned class for deserialization (and validation) and then the code should be really safe.
Thanks for clarification!

**Comment by andreea-anastasescu** [comment]
the changes were done - is anything else necessary here?

**Comment by RaduCStefanescu** [comment]
I will close this issue as it is fixed in #454 
Thank you @andreea-anastasescu !
